File: android/src/main/java/com/ccc/vcrtc_plugin/platform/ViewCollect.java
package com.ccc.vcrtc_plugin.platform;


import com.ccc.vcrtc_plugin.bean.Stream;
import com.vcrtc.VCRTCView;

import java.util.HashMap;
import java.util.Map;

public class ViewCollect {
    Map<String,VCRTCView> vcrtcViewMap;
    Map<String, Stream> vcrtcStreamMap;
    private static class SingletonClassInstance{
        private static final ViewCollect instance=new ViewCollect();
    }
    private ViewCollect(){
        vcrtcViewMap = new HashMap<String,VCRTCView>();
        vcrtcStreamMap = new HashMap<String,Stream>();
    }
    public static ViewCollect getInstance(){
        return SingletonClassInstance.instance;
    }
    public void setView(String uuid,VCRTCView view){
        this.vcrtcViewMap.put(uuid,view);
    }
    public Stream getStream(String uuid){
        return this.vcrtcStreamMap.get(uuid);
    }
    public void setStream(String uuid,Stream stream){
        this.vcrtcStreamMap.put(uuid,stream);
    }
    public VCRTCView getView(String uuid){
        return this.vcrtcViewMap.get(uuid);
    }
    public void removeView(String uuid){
        this.vcrtcViewMap.remove(uuid);
        this.vcrtcStreamMap.remove(uuid);
    }
}

File: ios/Classes/platform/ViewCollect.swift
//
//  ViewCollect.swift
//  vcrtc_plugin
//
//  Created by apple on 2021/1/19.
//
import VCRTC.VCVideoView
import Foundation
class ViewCollect {
    static let sharedInstance = ViewCollect()
    private init(){}
    private var viewMap = [String:VCVideoView]()
    public func setView(viewId:String,view:VCVideoView){
        viewMap[viewId] = view
    }
    public func getView(viewId:String)->VCVideoView{
        return viewMap[viewId]!
    }
    public func removeView(viewId:String){
        viewMap.remove(at: viewMap.index(forKey: viewId)!)
    }
}
